.github/workflows/main.yml

```diff
@@ -20,201 +20,143 @@ concurrency:
           name: installers-${{ runner.os }}-${{ github.sha }}-${{ github.run_id }}-${{ github.run_number }}-${{ github.run_attempt }}
-          path: examples_artifacts/
+          path: "${{ runner.temp }}/examples_artifacts"
           retention-days: 7
 
-  upload:
-    needs: package
-    runs-on: ubuntu-latest
-    if: github.event_name == 'push'
+  build:
+    name: Canary Build
+    needs: [tests]
+    # only build canary build if
+    # only build canary build if
+    # - prior steps succeeded,
+    # - this is the main repo, and
+    # - we are on the main (or feature) branch
+    if: >-
+      success()
+      && !github.event.repository.fork
+      && (
+        github.ref_name == 'main'
+        || startsWith(github.ref_name, 'feature/')
+      )
+    strategy:
+      matrix:
+        include:
+          - runner: ubuntu-latest
+            subdir: linux-64
+          - runner: macos-latest
+            subdir: osx-64
+          - runner: windows-latest
+            subdir: win-64
+    runs-on: ${{ matrix.runner }}
     steps:
-      - name: Retrieve the source code
-        uses: actions/checkout@v3
+      # Clean checkout of specific git ref needed for package metadata version
+      # which needs env vars GIT_DESCRIBE_TAG and GIT_BUILD_STR:
+      - uses: actions/checkout@v3
         with:
+          ref: ${{ github.ref }}
+          clean: true
           fetch-depth: 0
-      - name: Download the build artifacts
-        uses: actions/download-artifact@v3
-        with:
-          name: package-${{ github.sha }}
-          path: conda-bld
-      - name: Install conda packages
-        run: |
-          source $CONDA/bin/activate
-          conda install -y sphinx anaconda-client
-      - name: Upload to anaconda.org
-        env:
-          ANACONDA_TOKEN: ${{ secrets.ANACONDA_TOKEN }}
-          GITHUB_REF: ${{ github.ref }}
-        run: |
-          source $CONDA/bin/activate
-          [[ "$GITHUB_REF" =~ ^refs/tags/ ]] || export LABEL="--label dev"
-          anaconda --verbose --token $ANACONDA_TOKEN upload --user ctools $LABEL conda-bld/*/*.tar.bz2 --force
-
-  docs:
-    name: Check docs are up-to-date
-    runs-on: ubuntu-latest
-    steps:
-      - name: Retrieve the source code
-        uses: actions/checkout@v3
+      - name: Create and upload canary build
+        uses: conda/actions/canary-release@v23.3.0
         with:
-          fetch-depth: 0
-      - name: Install local constructor
-        run: |
-          source $CONDA/bin/activate
-          conda create -n constructor constructor jinja2
-          conda activate constructor
-          pip install -U . --no-deps
-      - name: Build docs
-        run: |
-          source $CONDA/bin/activate
-          conda activate constructor
-          set -ex
-          python scripts/make_docs.py
-          git diff --exit-code
+          package-name: ${{ github.event.repository.name }}
+          subdir: ${{ matrix.subdir }}
+          anaconda-org-channel: ctools
```

Comment:
Should we change to `conda-canary` here? xref https://github.com/conda/constructor/issues/612

Reply:
Still TODO?